python/tank/authentication/shotgun_shared/saml2_sso.py

```diff
@@ -0,0 +1,576 @@
+# Copyright (c) 2017 Autodesk.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+"""
+Module to support SSO login via a web browser and automated session renewal.
+
+This module, in conjunction with the slmodule, handles authentication on
+SSO enabled Shotgun websites. When we are in a session connected via SSO, we
+also handle automatic (and headless) renewal of the session.
+"""
+
+import base64
+import json
+import logging
+import os
+import time
+import urllib
+
+from .authentication_session_data import AuthenticationSessionData
+from ..ui.qt_abstraction import QtCore, QtNetwork, QtGui, QtWebKit
+from ... import LogManager
+
+log = LogManager.get_logger(__name__)
```

Comment:
Yes. This will give me time to do the RV and Toolkit refactor at the same time (I want the SSO core to be identical for both version, with custom application-specific glue code).